docs/api/web-contents.md

```diff
@@ -590,6 +590,25 @@ Emitted when `webContents` wants to do basic auth.
 
 The usage is the same with [the `login` event of `app`](app.md#event-login).
 
+#### Event: 'dialog'
+
+Returns:
+
+* `event` Event
+* `details` Object
+  * `messageText` string - Dialog message text.
+  * `defaultPromptText` string - Default prompt input text.
+  * `dialogType` string - 'alert' | 'confirm' | 'prompt'
```

Comment:
I think this should be the follow TS type (which you can generate via the right markdown syntax)

```
type DialogDetails = MessageDialogDetails | PromptDialogDetails;

type MessageDialogDetails = {
  message: string;
  type: 'alert' | 'confirm';
};

type PromptDialogDetails = {
  message?: string;
  defaultValue?: string;
  type: 'prompt';
}
```

Reply:
What is the right location and file to define this type?

Reply:
@vitekcerny this is done (as @MarshallOfSound said) via markdown syntax in the existing documentation. See https://github.com/electron/electron/blob/7acb513ba675aee12bdf52cf5ca8c3d162a93ef4/docs/api/content-tracing.md#L46 for a sort-of-similar example.

Reply:
Thank you for the link. I tried to follow the example and created 2 TS types: `MessageDialogDetails` and `PromptDialogDetails`.